bigcode_eval/evaluator.py

```diff
@@ -66,6 +66,7 @@ def generate_text(self, task_name, intermediate_generations=None):
 
         intermediate_save_generations_path = f"{os.path.splitext(self.args.save_generations_path)[0]}_{task_name}_intermediate.json"
         raw_generation_path = f"{self.args.save_results_dir}/{task_name}_raw_generations.json"
+        os.makedirs(self.args.save_results_dir, exist_ok=True)
```

Comment:
I think these calls are easier to debug later if you do it when you start actually writing. Otherwise, you can end up with an empty directory created even if no save results are requested. cc @meher-m

Reply:
Yea, makes sense. I will address this in the next PR.